### PR TITLE
[Eager] Refactor TensorAdd by template

### PR DESCRIPTION
**Pull request by veyron95**
### PR types
Others 

### PR changes
Others

### Describe
1. Refactor TensorAdd func  by template in gradient_accumulater.cc
2. Remove gradient_accumulation.c  and .h in eager/accumulation
3. Include gradient_accumulator.h instead of including gradient_accumulation.h

### CODE REVIEW

File: paddle/fluid/imperative/gradient_accumulator.cc [hunk 1]
```diff
@@ -214,9 +214,59 @@ void TensorAddImpl(const framework::Tensor& src, framework::Tensor* dst,
   func(dev_ctx, src, dst);
 }
 
-void TensorAdd(const framework::Variable& src, framework::Variable* dst) {
+template <typename VarType>
+std::shared_ptr<pten::DenseTensor> GetInnerDstTensor(VarType* dst) {
```
Comment (JiabinYang) [inline paddle/fluid/imperative/gradient_accumulator.cc]: use overload instead of template

**Review by JiabinYang — COMMENTED** [review]
some comment

File: paddle/fluid/imperative/gradient_accumulator.cc (hunk 1 above)
Comment (veyron95) [reply paddle/fluid/imperative/gradient_accumulator.cc]: Done, thx.

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**